### DISCUSSION TITLE
Help with Connect-PnPOnline Error The input is not a valid Base-64 string

### DISCUSSION THREAD

**Discussion by myatix** [pnp/powershell#2640] [Q&A]
Hi All,

I am having an issue connecting to using Connect-PnPOnline 

$tenant = $env:Tenant
$clientId = $env:ClientId
$certificateBase64Encoded = $env:Certificate
$templateLibUrl = $env:TemplateLibUrl

Connect-PnPOnline -Url $templateSiteUrl -Tenant $tenant -ClientId $clientId -CertificateBase64Encoded $certificateBase64Encoded

[Error]   ERROR: The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters.

My Certificate is stored in an Azure Key Vault and the $env:Certificate points to the Key Vault Certificate Identifier.

Can someone tell me why it returns this error?

**Comment by AndersRask** [comment]
It sounds like your base64 encoded cert isnt stored correctly in KV.

If you have access to KV you could verify the string. It should be a very long string that contain no whitespaces.
Usually it starts with something like MIIKcQIBAzCCCi0GCSqGSIb3DQEHA (example) and ends with =

**Reply by myatix** [reply]
@AndersRask Thanks for the response!

The certificate Identifier is https:/kv-xxxxx.vault.azure.net/certificates/<CertName>/GUID

![image](https://user-images.githubusercontent.com/48433526/207320385-a0f28005-2dc5-4bcf-b9e1-05a699780c34.png)

**Reply by myatix** [reply]
My Mistake...

This resolved the issue... :)

@Microsoft.KeyVault(SecretUri=https:/kv-xxxxx.vault.azure.net/certificates/GUID)
